components/status_dropdown/status_dropdown.tsx

```diff
@@ -63,35 +69,99 @@ export default class StatusDropdown extends React.PureComponent <Props, State> {
         },
     }
 
-    isUserOutOfOffice = (): boolean => {
-        return this.props.status === UserStatuses.OUT_OF_OFFICE;
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            openUp: false,
+            width: 0,
+        };
+    }
+
+    getCurrentDateTime = (tz: UserTimezone, enable: boolean): Date => {
+        let currentDate = new Date();
+        if (enable) {
+            if (tz.useAutomaticTimezone) {
+                currentDate = getCurrentDateTimeForTimezone(tz.automaticTimezone);
+            } else {
+                currentDate = getCurrentDateTimeForTimezone(tz.manualTimezone);
+            }
+        }
+        return currentDate;
     }
 
-    setStatus = (status: string): void => {
+    setStatus = (status: string, dndEndTime: any): void => {
```

Comment:
Do we not have a type for `dndEndTIme`? Looks like it can be `string` or `Date`. Any reason why it's both?